### PR TITLE
python3Packages.requests-gssapi: init at 1.2.3

### PR DESCRIPTION
**Pull request by JaviMerino**
## Description of changes

requests-gssapi lets you use Python Requests authenticated via GSSAPI. This PR adds the module at the latest version (1.2.3).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, in a couple of python applications. One of them is [koji](https://github.com/JaviMerino/nur-packages/blob/koji/pkgs/koji/default.nix)
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] ~~Tested basic functionality of all binary files (usually in `./result/bin/`)~~
  - It's a python module, so tested by using it an application instead.
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] ~~(Package updates) Added a release notes entry if the change is major or breaking~~
    - Not major nor breaking
  - [ ] ~~(Module updates) Added a release notes entry if the change is significant~~
    - Change is not significant
  - [ ] ~~(Module addition) Added a release notes entry if adding a new NixOS module~~
    - Not a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JaviMerino** [comment]
Thanks for trying it on darwin!

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ fetchPypi
+, lib
+, python3Packages
+}:
+python3Packages.buildPythonPackage rec {
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:5]:
```suggestion
}:

buildPythonPackage rec {
```

and add `buildPythonPackage` as input.

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  pname = "requests-gssapi";
+  version = "1.2.3";
+  src = fetchPypi {
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:8]: Upstream says Python > [3.8](https://github.com/pythongssapi/requests-gssapi/blob/main/pyproject.toml#L11).

```suggestion
  version = "1.2.3";
  pyproject = true;

  disabled = pythonOlder "3.8";

  src = fetchPypi {
```

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    inherit pname version;
+    hash = "sha256-IHhFCJgUAfcVPJM+7QlTOJM6QIGNplolnb8tgNzLFQ4=";
+  };
+
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:12]:
```suggestion
  };

  nativeBuildInputs = [
    setuptools
  ];

```

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  propagatedBuildInputs = with python3Packages; [
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:13]:
```suggestion
  propagatedBuildInputs = [
```

see above.

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    gssapi
+    requests
+  ];
+  pythonImportCheck = [ "requests_gssapi" ];
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:17]: Upstream uses `pytest` for the unit tests.

```suggestion
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];

  pythonImportCheck = [ "requests_gssapi" ];
```

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "A GSSAPI authentication handler for python-requests";
+    homepage = "https://github.com/pythongssapi/requests-gssapi";
+    maintainers = with maintainers; [ javimerino ];
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:22]: Tools (e.g., `nix-update`) can add a link to the changelog if the changelog entry is present.

```suggestion
    homepage = "https://github.com/pythongssapi/requests-gssapi";
    changelog = "https://github.com/pythongssapi/requests-gssapi/blob/v${version}/HISTORY.rst";
    maintainers = with maintainers; [ javimerino ];
```

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    license = [ licenses.isc ];
+    platforms = platforms.all;
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:24]: This is the default.

```suggestion
```

File: pkgs/development/python-modules/requests-gssapi/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+}
```
Comment (fabaff) [inline pkgs/development/python-modules/requests-gssapi/default.nix:27]:
```suggestion
  };
}
```

File: pkgs/development/python-modules/requests-gssapi/default.nix (hunk 6 above)
Comment (JaviMerino) [reply pkgs/development/python-modules/requests-gssapi/default.nix]: Today I learned! Thanks

**Comment by JaviMerino** [comment]
Thanks for the review @fabaff , I have applied all your suggestions

**Comment by JaviMerino** [comment]
@fabaff I think I solved all the issues, is there anything else I have to do to have it approved?

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 288070` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.requests-gssapi</li>
    <li>python311Packages.requests-gssapi.dist</li>
    <li>python312Packages.requests-gssapi</li>
    <li>python312Packages.requests-gssapi.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Looks good to me.

**Pull request merged**